ports/libmlservo/src/lib.rs

```diff
@@ -110,28 +119,23 @@ pub unsafe extern "C" fn discard_servo(servo: ServoInstance) {
 }
 
 struct WindowInstance {
+    ctxt: EGLContext,
+    surf: EGLSurface,
+    disp: EGLDisplay,
     gl: Rc<Gl>,
     width: u32,
     height: u32,
     hidpi: f32,
 }
 
-impl WindowInstance {
-    fn new(gl: Rc<Gl>, width: u32, height: u32, hidpi: f32) -> WindowInstance {
-        WindowInstance {
-            gl: gl,
-            width: width,
-            height: height,
-            hidpi: hidpi,
-        }
-    }
-}
-
 impl WindowMethods for WindowInstance {
     fn present(&self) {
+        SwapBuffers(self.disp, self.surf);
     }
 
     fn prepare_for_composite(&self, _w: Length<u32, DevicePixel>, _h: Length<u32, DevicePixel>) -> bool {
+        MakeCurrent(self.disp, self.surf, self.surf, self.ctxt);
+        self.gl.viewport(0, 0, self.width as i32, self.height as i32);
```

Comment:
Shouldn't this be using the w and h arguments instead?

Reply:
Don't know. On sample runs they agree, except on the very first call where `_w` and `_h` are zero. I think the GL viewport has to agree with the viewport reported by `get_coordinates()` which uses `self.x` and `self.y`.

Reply:
The very first call is just to make sure we call make_current early: https://github.com/servo/servo/pull/16903